x/ibc/03-connection/keeper/handshake_test.go

```diff
@@ -47,92 +46,93 @@ func (suite *KeeperTestSuite) TestConnOpenInit() {
 			suite.chainB.updateClient(suite.chainA)
+			return 0
 		}, false},
-		{"consensus state verification invalid", ibctypes.ValidProof{}, ibctypes.InvalidProof{}, func() {
+		{"consensus state verification invalid", func() uint64 {
 			suite.chainB.CreateClient(suite.chainA)
 			suite.chainA.CreateClient(suite.chainB)
-			consensusHeight = suite.chainB.Header.Height
 			suite.chainA.createConnection(testConnectionIDA, testConnectionIDB, testClientIDB, testClientIDA, exported.INIT)
 			suite.chainB.updateClient(suite.chainA)
 			suite.chainA.updateClient(suite.chainB)
+			return uint64(suite.chainB.Header.Height)
 		}, false},
-		{"invalid previous connection", ibctypes.ValidProof{}, ibctypes.ValidProof{}, func() {
+		{"invalid previous connection", func() uint64 {
 			suite.chainB.CreateClient(suite.chainA)
 			suite.chainA.CreateClient(suite.chainB)
-			consensusHeight = suite.chainB.Header.Height
 			suite.chainB.createConnection(testConnectionIDB, testConnectionIDA, testClientIDA, testClientIDB, exported.UNINITIALIZED)
 			suite.chainB.updateClient(suite.chainA)
 			suite.chainA.updateClient(suite.chainB)
+			return 0
 		}, false},
-		{"couldn't add connection to client", ibctypes.ValidProof{}, ibctypes.ValidProof{}, func() {
+		{"couldn't add connection to client", func() uint64 {
 			suite.chainB.CreateClient(suite.chainA)
-			consensusHeight = suite.chainB.Header.Height
 			suite.chainA.createConnection(testConnectionIDB, testConnectionIDA, testClientIDB, testClientIDA, exported.UNINITIALIZED)
 			suite.chainB.updateClient(suite.chainA)
 			suite.chainA.updateClient(suite.chainB)
+			return 0
 		}, false},
 	}
 
 	for i, tc := range testCases {
+		// TODO remove me, just for fewer printouts
+		if i > 0 {
+			continue
+		}
+		// end TODO
 		tc := tc
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
 			suite.SetupTest() // reset
 
-			tc.malleate()
+			consensusHeight := tc.malleate()
 
-			// connectionKey := ibctypes.KeyConnection(testConnectionIDA)
-			// proofInit, proofHeight := suite.queryProof(connectionKey)
+			connectionKey := ibctypes.KeyConnection(testConnectionIDA)
+			proofInit, proofHeight := queryProof(suite.chainA, connectionKey)
 
-			// consensusKey := ibctypes.KeyConsensusState(testClientIDA, uint64(proofHeight))
-			// proofConsensus, consensusHeight := suite.queryProof(consensusKey)
+			consensusKey := ibctypes.KeyConsensusState(testClientIDB, consensusHeight)
+			proofConsensus, _ := queryProof(suite.chainA, consensusKey)
 
 			err := suite.chainB.App.IBCKeeper.ConnectionKeeper.ConnOpenTry(
 				suite.chainB.GetContext(), testConnectionIDB, counterparty, testClientIDA,
-				connection.GetCompatibleVersions(), tc.proofInit, tc.proofConsensus,
-				uint64(proofHeight), uint64(consensusHeight),
+				connection.GetCompatibleVersions(), proofInit, proofConsensus,
+				uint64(proofHeight+1), consensusHeight,
```

Comment:
We need to note this +1 offset in the relayer - the app hash for height `n` is stored in header `n + 1`.

Reply:
https://github.com/cosmos/relayer/issues/39